x-pack/plugins/rollup/__jest__/integration_tests/job_create.test.js

```diff
@@ -0,0 +1,1298 @@
+
+          const errorMessages = getFormErrorsMessages();
+          expect(errorMessages).toContain('Index pattern cannot have the same as the rollup index.');
+          expect(errorMessages).toContain('Rollup index cannot have the same as the index pattern.');
+        });
+      });
+
+      describe('rollup index name', () => {
+        beforeEach(() => {
+          expect(findTestSubject('rollupJobNextButton').props().disabled).toBe(false);
+        });
+
+        afterEach(() => {
+          expect(findTestSubject('rollupJobNextButton').props().disabled).toBe(true);
+        });
+
+        it('should not allow spaces', () => {
+          form.setInputValue('rollupIndexName', 'with space');
+          userActions.clickNextStep();
+          expect(getFormErrorsMessages()).toContain('Remove the spaces from your rollup index name.');
+        });
+
+        it('should not allow invalid characters', () => {
+          const expectInvalidChar = (char) => {
+            form.setInputValue('rollupIndexName', `rollup_index_${char}`);
+            userActions.clickNextStep();
+            expect(getFormErrorsMessages()).toContain(`Remove the characters ${char} from your rollup index name.`);
+          };
+
+          [...INDEX_PATTERN_ILLEGAL_CHARACTERS_VISIBLE, ','].reduce((promise, char) => {
+            return promise.then(() => expectInvalidChar(char));
+          }, Promise.resolve());
+        });
+
+        it('should not allow a dot as first character', () => {
+          form.setInputValue('rollupIndexName', '.kibana');
+          userActions.clickNextStep();
+          expect(getFormErrorsMessages()).toContain('Index names cannot begin with periods.');
+        });
+      });
+
+      describe('rollup cron', () => {
+        const changeFrequency = (value) => {
+          findTestSubject('rollupJobCreateFrequencySelect').simulate('change', { target: { value } });
+        };
+
+        const generateStringSequenceOfNumbers = (total) => (
+          new Array(total).fill('').map((_, i) => i < 10 ? `0${i}` : i.toString())
+        );
+
+        describe('frequency', () => {
+          it('should allow "minute", "hour", "day", "week", "month", "year"', () => {
+            const frequencySelect = findTestSubject('rollupJobCreateFrequencySelect');
+            const options = frequencySelect.find('option').map(option => option.text());
+            expect(options).toEqual(['minute', 'hour', 'day', 'week', 'month', 'year']);
+          });
+
+          describe('every minute', () => {
+            it('should not have any additional configuration', () => {
+              changeFrequency(MINUTE);
+              expect(findTestSubject('rollupCronFrequencyConfiguration').length).toBe(0);
+            });
+          });
+
+          describe('hourly', () => {
+            beforeEach(() => {
+              changeFrequency(HOUR);
+            });
+
+            it('should have 1 additional configuration', () => {
```

Comment:
I've found that a neat way to shorten descriptions and make them less repetitive is to omit the word "should". For example, this could be "has 1 additional condition". When it passes, you know that it does, and when it fails you know that it doesn't -- you get the same information, but without seeing "should X, should Y" repeated dozens of times. 🙂

Reply:
I'll take it into account for future tests. Although to me it's only a `tit + TAB` shortcut command for the block prepopulated with `'should '`.